apps/system/js/input_window_manager.js

```diff
@@ -269,14 +269,7 @@
   };
 
   InputWindowManager.prototype._getMemory = function iwm_getMemory() {
-    if ('getFeature' in navigator) {
-      navigator.getFeature('hardware.memory').then(mem => {
-        this._totalMemory = mem;
-      }, () => {
-        console.error('InputWindowManager: ' +
-          'Failed to retrieve total memory of the device.');
-      });
-    }
+    this._totalMemory = Service.query('getDeviceMemory');
```

Comment:
nit: This may cause a problem when being async. I don't have a good solution right now, but maybe we should keep the module and figure out how to resolve the asynchronous problem before we switch to the query here.